### PR TITLE
8290399: [macos] Aqua LAF does not fire an action event if combo box menu is displayed

### PR DESCRIPTION
**Pull request by prsadhuk**
When the user types Enter in an editable combo box, Aqua LAF fails to fire an action event if the menu is displayed.
It seems actionEvent is fired when popup menu is not visible. 
Fix is to make sure ActionEvent is fired irrespective of popup visiblity as is done in other L&F 
(for ex, in MetalL&F it calls "ENTER", JTextField.notifyAction, where it calls postActionEvent irrespectively)
https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/javax/swing/JTextField.java#L856

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8290399](https://bugs.openjdk.org/browse/JDK-8290399): [macos] Aqua LAF does not fire an action event if combo box menu is displayed

### Reviewers
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9551/head:pull/9551` \
`$ git checkout pull/9551`

Update a local copy of the PR: \
`$ git checkout pull/9551` \
`$ git pull https://git.openjdk.org/jdk pull/9551/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9551`

View PR using the GUI difftool: \
`$ git pr show -t 9551`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9551.diff">https://git.openjdk.org/jdk/pull/9551.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/swing/JComboBox/JComboBoxActionEvent.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        UIManager.setLookAndFeel("com.apple.laf.AquaLookAndFeel");
+
+        PassFailJFrame pfjFrame = new PassFailJFrame("JScrollPane "
+                + "Test Instructions", instructionsText, 5);
+
+        createAndShowGUI();
+
+        pfjFrame.awaitAndCheck();
+    }
+}
```
Comment (prsadhuk) [inline test/jdk/javax/swing/JComboBox/JComboBoxActionEvent.java:91]: Tried automating the test but it seems to call BasicComboBoxUI#focusLost when ENTER key is pressed, which calls actionPerformed thus defeating the purpose.


```

public class TestEditableComboBox
{
    private static JFrame fr;
    private static JComboBox<String> cb;
    private static Point loc;
    private static Dimension dim;
    private static volatile boolean actionDone = false;

    public TestEditableComboBox()
    {
        // Instructions:
        // Click the arrow to display the menu.
        // While the menu is displayed, edit the text to create a new value.
        // Type return.
        // The action listener should be invoked.

        // Using Aqua LAF, the return dismisses the menu but does invoke action listeners.
        // A second return is needed to invoke actions listeners.
        // This behavior is inconsistent with other LAFs and native NSComboBox.

//        if (true) {
//            try {
//                //UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
//                UIManager.setLookAndFeel("com.apple.laf.AquaLookAndFeel");
//            } catch (Exception e) {
//                System.err.println(e.toString());
//            }
//        }

        fr = new JFrame("Test Editable Combo Box");
        String[] values = new String[] { "Apple", "Orange", "Pear" };
        cb = new JComboBox<>(values);
        cb.setEditable(true);
        cb.addActionListener(e -> {
            //actionDone = true;
                System.err.println("Action listener called: " + e.getActionCommand());
                });

        Container c = fr.getContentPane();
        c.setLayout(new FlowLayout());
        c.add(cb);
        fr.pack();
        //fr.setBounds(400, 100, 400, 80);
        fr.setLocationRelativeTo(null);
        fr.setVisible(true);
    }

    public static void main(String[] args) throws Exception {

        java.util.List<String> lafs = Arrays.stream(UIManager.getInstalledLookAndFeels())
                                  .map(UIManager.LookAndFeelInfo::getClassName)
                                  .collect(Collectors.toList());
        for (final String laf : lafs) {
            try {
                try {
                    UIManager.setLookAndFeel(laf);
                    System.out.println("Testing L&F: " + laf);
                } catch (UnsupportedLookAndFeelException e) {
                    System.out.println("Skipping unsupported L&F: " + laf);
                }
                SwingUtilities.invokeAndWait(TestEditableComboBox::new);
                SwingUtilities.invokeAndWait(() -> {
                    loc = cb.getLocationOnScreen();
                    dim = cb.getSize();
                });
                Robot robot = new Robot();
                robot.waitForIdle();
                robot.delay(1000);
                robot.mouseMove(loc.x + dim.width - 2, loc.y);
                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
                robot.waitForIdle();
                robot.delay(1000);
                robot.keyPress(KeyEvent.VK_A);
                robot.keyRelease(KeyEvent.VK_A);
                robot.waitForIdle();
                robot.delay(1000);
                robot.keyPress(KeyEvent.VK_ENTER);
                robot.keyRelease(KeyEvent.VK_ENTER);
                robot.waitForIdle();
                robot.delay(1000);
//                if (!actionDone) {
//                    throw new RuntimeException("ActionEvent not fired");
//                }
            } finally {
                SwingUtilities.invokeAndWait(() -> {
                    if (fr != null) {
                        fr.dispose();
                    }
                });
            }
        }
    }
}

```

File: test/jdk/javax/swing/JComboBox/JComboBoxActionEvent.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+            PassFailJFrame.addTestWindow(frame);
+            PassFailJFrame.positionTestWindow(frame,
+                    PassFailJFrame.Position.HORIZONTAL);
+        });
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        UIManager.setLookAndFeel("com.apple.laf.AquaLookAndFeel");
```
Comment (azvegint) [inline test/jdk/javax/swing/JComboBox/JComboBoxActionEvent.java:82]: Shouldn't the test be macOS only?

Comment (prsadhuk) [reply test/jdk/javax/swing/JComboBox/JComboBoxActionEvent.java:83]: done

**Review by azvegint — APPROVED** [review]

**Review by prrace — APPROVED** [review]
OK .. although the existing code even if it is surprising, looks very deliberate .. which makes me a little leery.
But we've plenty of time in JDK 20 to see if this causes a problem.

**Pull request closed without merging**